zoo/repos/github.py

```
@@ -79,6 +81,14 @@ def get_languages(remote_id):
     return langs_percent
 
 
+def get_file_content(github_id, path, ref="master"):
+    try:
+        proj = get_project(github_id)
+        return decode(proj.get_contents(path, ref).raw_data["content"])
+    except Exception as e:
+        raise FileNotFoundError(e)
```

Is it a good idea to catch all exceptions and reraise as `FileNotFoundError`? I'm just not sure about it, I would probably catch only the relevant exceptions.

---

No it is not a good idea to have a catch all clause. I just couldn't find documentation on what kind of exception the library throws. I will look at the source code and change it accordingly.

---

So the fix here is to just catch RepositoryNotFoundError error and let it fail on all other exceptions.

---

BTW you can also use `raise SomeError from exc`
see https://stackoverflow.com/questions/24752395/python-raise-from-usage